### DISCUSSION TITLE
new positional argument, init, for Single DB mode

### DISCUSSION THREAD

**Discussion by minghungcho** [littleK0i/SnowDDL#13] [Ideas]
SnowDDL is an excellent tool to help Snowflake developers to manage a Snowflake Database Project. Similar to [dbt init](https://docs.getdbt.com/reference/commands/init), it will be great if a developer can use SnowDDL to initialize a folder structure for a new Snowflake Database project.

The **init** command will prompt for database project name (served as entry point to config directory), and a database name. By default, it also generate a public schema folder with a list of sub-folders (table, view, procedure, sequence, ..) for different objects. This will help developers quickly understand where to place objects.

I'd recommend to also add params.yaml under public folder with is_sandbox: true by default. This provides a safe option not to disrupt an existing database.

Lastly, at the end of init command, maybe SnowDDL can display a sample CLI. So user can copy to try out later.

**Comment by littleK0i** [comment]
It's a good idea, I'll definitely implement it at some point in near future.

Most likely, it will be a separate command, e.g. `snowddl-init`, due to the way how `argparse` handles sub-commands. Separate commands are a bit easier to manage in terms of code complexity.

Thank you.
